nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -480,6 +519,11 @@ in {
               'trusted_proxies' => ${writePhpArrary (c.trustedProxies)},
               ${optionalString (c.defaultPhoneRegion != null) "'default_phone_region' => '${c.defaultPhoneRegion}',"}
             ];
+
+            $EXTRACONFIG = json_decode('${builtins.toJSON cfg.extraOptions}', true);
```

Comment:
These should probably be read from a file (though writing it to the store is fine here) as well, as the JSON may contain single-quote characters.

Reply:
implemented that it https://github.com/NixOS/nixpkgs/pull/118093/commits/83a669a0bef463b3202a544b33efc9109605ce83

Reply:
Could you specify what you mean with RFC42-style approach? As far as I can see it suggests doing these extra-options like hooks as structured attribute sets instead of strings, which is what I have done (via json formatting, since I couldn't find any better way to serialise to php).

Apart from that, the rfc talks a lot about balancing these options with "normal" nixos module options, but I didn't really want to replace any of the other parts, since the whole module is quiet complex and I don't actually understand any php (I just needed a way for secrets & extra options in my own deployment). I guess we could rename `extraOptions` to `extraSettings` though.